### PR TITLE
Refactor return statement of constraint_macro

### PR DESCRIPTION
**Pull request by blegat**
Mirror the change made in https://github.com/JuliaOpt/JuMP.jl/pull/1315/commits/365d4915cebf507a48d8e556b15c3e854434f71b for the `constraint_macro`

### CODE REVIEW

File: src/macros.jl [hunk 1]
```diff
@@ -446,19 +446,28 @@ function constraint_macro(args, macro_name::Symbol, parsefun::Function)
...
+    if anonvar
+        # Anonymous constraint, no need to register it to a name
+        # nor to assign it to a variable in the user scope.
+        # We simply return the constraint reference
+        assignmentcode = variable
+    else
+        # We register the constraint reference to its name and
+        # we assign it to a variable of this name
+        assignmentcode = quote
+            registervar($m, $quotvarname, $variable)
```
Comment (mlubin) [inline src/macros.jl]: `registercon`?

File: src/macros.jl [hunk 2]
```diff
@@ -446,19 +446,28 @@ function constraint_macro(args, macro_name::Symbol, parsefun::Function)
         $parsecode
         $(refcall) = $constraintcall
     end
+
     # Determine the return type of addconstraint. This is needed for JuMP extensions for which this is different than ConstraintRef
     contype = :( constrainttype($m) )
+    creationcode = getloopedcode(variable, code, condition, idxvars, idxsets, contype, requestedcontainer)
+
+    if anonvar
+        # Anonymous constraint, no need to register it to a name
```
Comment (mlubin) [inline src/macros.jl]: Clarify that "name" here refers to the model-level dictionary and not the string name.

File: src/macros.jl [hunk 3]
```diff
@@ -446,19 +446,28 @@ function constraint_macro(args, macro_name::Symbol, parsefun::Function)
...
+        # nor to assign it to a variable in the user scope.
+        # We simply return the constraint reference
+        assignmentcode = variable
+    else
+        # We register the constraint reference to its name and
+        # we assign it to a variable of this name
```
Comment (mlubin) [inline src/macros.jl]: to a variable in the local scope

File: src/macros.jl (hunk 1 above)
Comment (blegat) [reply src/macros.jl]: Wow, weird that it passes the tests

Comment (blegat) [reply src/macros.jl]: Ah only the error message changes so it makes sense.

File: src/macros.jl [hunk 4]
```diff
@@ -446,19 +446,28 @@ function constraint_macro(args, macro_name::Symbol, parsefun::Function)
...
     contype = :( constrainttype($m) )
+    creationcode = getloopedcode(variable, code, condition, idxvars, idxsets, contype, requestedcontainer)
+
+    if anonvar
+        # Anonymous constraint, no need to register it in the model-level
+        # dictionary nor to assign it to a variable in the user scope.
+        # We simply return the constraint reference
+        assignmentcode = variable
+    else
+        # We register the constraint reference to its name and
```
Comment (mlubin) [inline src/macros.jl]: We register the constraint reference to its name in the model-level dictionary

**Pull request merged**